src/firefly/java/edu/caltech/ipac/firefly/server/db/DbAdapter.java

```diff
@@ -20,6 +21,25 @@
     String SQLITE = "sqlite";
     String HSQL = "hsql";
 
+    /*
+      CLEAN UP POLICY:
+        A rough estimate: A search results of one million rows displayed in the triview(chart + image + table) takes around 500 MB
+        There are two stages of clean-up; compact(remove all temp tables), then shutdown(remove from memory)
+
+        - Compact DB once it has idled longer than CLEANUP_INTVL
+        - Shutdown DB once it has expired; idle longer than MAX_IDLE_TIME
+        - Shutdown DB based on LRU(Least Recently Used) once the total rows have exceeded MAX_MEMORY_ROWS
+
+        Current settings:
+          - CLEANUP_INTVL:  1 minutes
+          - MAX_IDLE_TIME: 15 minutes
+          - MAX_MEMORY_ROWS:  250k rows for every 1GB of max heap, between the range of 1-10 millions.
+     */
+    long MAX_IDLE_TIME  = 1000 * 60 * 15;   // will shutdown database if idle more than 15 minutes.
+    int  CLEANUP_INTVL  = 1000 * 60;        // check every 1 minutes
+    long MAX_MEMORY_ROWS = Math.min(10000000, Math.max(1000000, Runtime.getRuntime().maxMemory()/1024/1024/1024 * 250000));
+
```

Comment:
So yes, I'll have to make this a server's attribute instead of an application one, similar to VIS_MEM_SHARE_MEM.  However, I think it's should be a fixed value rather than on available memory.

Status page link is posted in PR.  http://localhost:8080/firefly/admin/status

Reply:
I've changed my mind.  The simplest solution is to set MAX_MEMORY_ROWS based on available memory.  That way, it does not matter how many applications are deployed in the same jvm.  I'll use the same 250k/GB of memory but based on available size instead.  This performance optimization.  It should work even if nothing were in memory.  What do you think?

Reply:
I think that is fine, I like basing it on  available memory